### PR TITLE
Explore: Remove exploreMixedDatasource feature toggle

### LINKED ISSUE #52417: Explore: Remove feature flag for mixed datasources
As part of the work for #44438, we added a feature flag to enable the mixed datasource view for explore that is defaulted to off. 

When the following is done, remove the feature flag to enable it for everyone:

- We are confident the mixed datasource query importing is as we decide (this is currrently being discussed)
- The logs and traces squad has looked into how results are displayed, both in terms of the log volume and logs lines panels (see #33120)
- Ensure all the features are working correctly with multiple data sources (e.g. logs volume, logs context). If there's anything missing to get feature parity consider if we could document/advertise it as "beta/experimental/ongoing" and incrementally close the gap before mixed data sources become the only option.

This list may be added to, as work on the mixed datasource is ongoing.

Additionally, evaluate feedback on the query importing that we get and decide if we want to follow normal import logic or remove it.
```[tasklist]
- [ ] https://github.com/grafana/grafana/issues/65360
- [x] Enable in ops (internally)
- [ ] https://github.com/grafana/grafana/issues/66396
- [ ] Check documentation to make sure it makes sense
```

### PR DESCRIPTION
**Pull request by Elfo404**
**What is this feature?**

Removes the `exploreMixedDatasource` feature flag. The flag was enabled by default in 10.0.0 and no issues has been reported about it

**Which issue(s) does this PR fix?**:

Fixes #52417 

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

a couple of now irrelevant tests have been removed. Please test that:
- Navigating to explore with mixed URL works
- Query History works as expected
- adding/removing query rows works as expected

### CODE REVIEW

**Review by harisrozajac — APPROVED** [review]
Looks great to me 👍

File: public/app/core/utils/explore.ts [hunk 1]
```diff
@@ -75,21 +75,6 @@ export async function getExploreUrl(args: GetExploreUrlArguments): Promise<strin
     .filter((t) => t.datasource?.uid !== ExpressionDatasourceUID);
   let url: string | undefined;
   // if the mixed datasource is not enabled for explore, choose only one datasource
```
Comment (ifrost) [inline public/app/core/utils/explore.ts:77]:
```suggestion
```

**Review by ifrost — APPROVED** [review]

File: public/app/features/explore/hooks/useStateSync/index.ts
Comment (gelicia) [inline public/app/features/explore/hooks/useStateSync/index.ts:1]: There's a couple mentions of `allowMIxed` in this that I can't make suggested changes on bc they're too far from the other changes.

**Review by gelicia — APPROVED** [review]

**Comment by Elfo404** [comment]
i should have removed all the leftovers, merging this, if i missed something else i'll create a follow up PR. thanks for the catches @gelicia @ifrost

**Pull request merged**